### ISSUE TITLE
Configured resolvers arent always used.

### ISSUE THREAD

**Issue by kmpm** [caddyserver/certmagic#132]
## What version of the package are you using?
a1d00125a60032bae3bbbea756ea637d7105c568

## What are you trying to do?
I am using certmagic through caddy, while in a network that uses split-head DNS, to get a certificate using the Loopia DNS plugin.
Resolvers are configured to use the external registrar DNS and that works fine on Windows but
whenever I tried to do the same on Linux it fails to get a certificate. When I moved the Linux machine off to a network that wasn't affected by this split-head DNS it worked.
This isn't affected by the Loopia DNS plugin per se but the fact that caddy cant find the SOA for the domain it tries to validate and then on Linux falls back to using whatever DNS-server it finds in /etc/resolve.conf, regardles of what I configure in the `resolvers` value in the Caddyfile or JSON config.

## What steps did you take?
Don't really know how to simulate this without setting up some DNS-stuff.
I found out that it happens by using tcpdump and logging the DNS traffic while caddy tried to get a certificate.
Then by plainly looking at the source code for certmagic.

## What did you expect to happen, and what actually happened instead?
I expected caddy / certmagic to use the external DNS-server(s) I had configured as resolvers.
It ended up using the internal DNS-servers it found in `/etc/resolve.conf`

## How do you think this should be fixed?
I made a change in PR #131 that lets you add the string `!important`as either a suffix or separate entry to the DNS-server in the configured resolvers. 
If that is not accepted I need a way to tell certmagic (through caddy) to use the DNS servers I configure and nothing else.

## Please link to any related issues, pull requests, and/or discussion
- https://caddy.community/t/windows-linux-amd64-validates-acme-dns-but-raspberry-os-armv7l-does-not/12595/9
- PR #131

## Bonus: What do you use CertMagic for, and do you find it useful?
I use it with caddy and I find it really good and useful. Who knows if I end up using it in some other fashion. Either way I am happy that I have found out about it.

**Comment by mholt** [comment]
Thanks for the PR. Gonna be honest I hate the `!important` thing. I wonder if there's ever any good reason to use the fallback resolvers when the user explicitly configures certain resolvers. If not, maybe we should just make that always be the case (use only user-configured resolvers). What do you think?

**Comment by kmpm** [comment]
I hate it to but I couldn't figure out a way to make it backwards compatible in a simpler way.
I'd prefer to have it use whatever resolvers that is configured and nothing else, no fallback. 

I can create another PR if you want to.

**Comment by mholt** [comment]
Yeah, I think it's unexpected to use resolvers that aren't configured. That is probably a bug (and I think a remnant of how lego works, from when we borrowed the DNS logic from there). If resolvers are explicitly configured, only those should be used.

So yeah, if you want to update your PR to do that instead, that'd probably be good! :+1:

**Comment by kmpm** [comment]
I don't have time today, but I'll fix it tomorrow morning ( UTC+1).

**Comment by mholt** [comment]
Sure, thanks so much! No hurry.
